libinteract/libinteract.py

```diff
@@ -1457,210 +1460,244 @@ def parse_hbs_file(fname):
+            hb_occur = (float(hb_count) / float(num_frames)) * 100
+            
             # Convert the identifier from a frozenset to a list
             # to get items by index (the order does not matter)
             identifier_as_list = list(identifier)
+            
             # Get info about the first residue
             res1 = identifier_as_list[0]
+
+            # The index is taken from the 'uni_identifiers' because
+            # the index of the residue in the topology may be different
+            # from the index found in the reference PDB structure
             res1_resix = uni_id2ix[res1]
             res1_segid, res1_resid = res1[:2]
+            
             # Get info about the second residue (if present)
             res2 = \
                 res1 if len(identifier) == 1 else identifier_as_list[1]
+
+            # The index is taken from the 'uni_identifiers' because
+            # the index of the residue in the topology may be different
+            # from the index found in the reference PDB structure
             res2_resix = uni_id2ix[res2]
             res2_segid, res2_resid = res2[:2]
+            
             # Fill the full matrix if requested
             if do_fullmatrix:
-                fullmatrix[res1_resix, res2_resix] = hb_pers
-                fullmatrix[res2_resix, res1_resix] = hb_pers
-            # Format the output string if requesets
-            if perresidue:
-                if hb_pers > perco:
-                    outstr += outstr_fmt.format(res1_segid, res1_resid,
-                                                res1_segid, res1_resid,
-                                                hb_pers)
-    
-    # Do not merge hydrogen bonds per residue
-    if not perresidue:
+                full_matrix[res1_resix, res2_resix] = hb_occur
+                full_matrix[res2_resix, res1_resix] = hb_occur
         
-        # Utility function to get the identifier of a hydrogen bond
-        get_list_identifier = \
-            lambda uni, hbond: [(uni.atoms[hbond[0]].segid,
-                                 uni.atoms[hbond[0]].resid,
-                                 uni.atoms[hbond[0]].resname,
-                                 "residue"),
-                                (uni.atoms[hbond[1]].segid,
-                                 uni.atoms[hbond[1]].resid,
-                                 uni.atoms[hbond[1]].resname,
-                                 "residue")]
-
-        # Count hydrogen bonds by type
-        table = h.count_by_type()
-        hbonds_identifiers = \
-            [get_list_identifier(uni, hbond) for hbond in table]
-        
-        # For each hydrogen bonds identified
-        for i, hbidentifier in enumerate(hbonds_identifiers):
+    # Get the hydrogen bonds' counts (not using the count_by_ids()
+    # method of the HydrogenBondAnalysis class since there may be
+    # a bug when reporting atom indexes)
+    counts = collections.Counter(\
```

Comment:
this + line 1653 looks very similar to what we did in line 1606. If it's possible it would make sense to do this once and reuse it rather than run it twice

Reply:
They are similar, but they refer to different data in the output data structure that MDAnalysis gives at the end of its hydrogen bonds analysis.

Reply:
got it! Thanks